src/PygmentSharp/PygmentSharp.Core/Lexing/JavaLexer.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using PygmentSharp.Core.Tokens;
+
+namespace PygmentSharp.Core.Lexing
+{
+
+    /// <summary>
+    /// A lexer for Java
+    /// </summary>
+    [Lexer("Java", AlternateNames = "java")]
+    [LexerFileExtension("*.java")]
+    public class JavaLexer : RegexLexer
+    {
+        /// <summary>
+        /// Gets the state transition rules for the lexer. Each time a regex is matched,
+        /// the internal state machine can be bumped to a new state which determines what
+        /// regexes become valid again
+        /// </summary>
+        /// <returns></returns>
+        protected override IDictionary<string, StateRule[]> GetStateRules()
+        {
+
+            System.Console.WriteLine("Using java lexer");
+            /*
+             * what to do about this line:

```

Comment:
Was this a question for me, or a note-to-self?

I think you've setup the DefaultRegexOptions in a way that is consistent with these python rules. But tbh, im not a python guy; and most of my exposure to it has been slowly learning what I needed to know to port this package :)